crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -396,58 +232,73 @@ fn finalize_operator_pending_transfers<T: Config>(
             return Err(TransitionError::OperatorNotRegistered);
         }
 
-        let mut total_stake = operator
+        // if there are no deposits, withdrawls, and epoch rewards for this operator
+        // then short-circuit and return early.
+        if operator.deposits_in_epoch.is_zero()
+            && operator.withdrawals_in_epoch.is_zero()
+            && operator.current_epoch_rewards.is_zero()
+        {
+            return Ok(operator.current_total_stake);
+        }
+
+        let total_stake = operator
             .current_total_stake
             .checked_add(&operator.current_epoch_rewards)
             .ok_or(TransitionError::BalanceOverflow)?;
 
-        let mut total_shares = operator.total_shares;
-        finalize_pending_withdrawals::<T>(
-            operator.current_domain_id,
-            operator_id,
-            &mut total_stake,
-            &mut total_shares,
-            domain_block_number,
-        )?;
+        let total_shares = operator.current_total_shares;
 
-        finalize_pending_deposits::<T>(operator_id, &mut total_stake, &mut total_shares)?;
+        let share_price = SharePrice::new::<T>(total_shares, total_stake);
+
+        // calculate and subtract total withdrew shares from previous epoch
+        let (total_stake, total_shares) = if !operator.withdrawals_in_epoch.is_zero() {
```

Comment:
If we can make `total_stake` and `total_shares` mut then we won't need the `if else`, non-blocker